rpc/src/v1/impls/eth.rs

```diff
@@ -209,6 +215,43 @@ impl Eth for EthClient {
 				to_value(&logs)
 			})
 	}
+
+	fn work(&self, params: Params) -> Result<Value, Error> {
+		match params {
+			Params::None => {
+				let c = take_weak!(self.client);
+				let u = c.sealing_block().lock().unwrap();
+				match *u {
+					Some(ref b) => {
+						let pow_hash = b.hash();
+						let target = Ethash::difficulty_to_boundary(b.block().header().difficulty());
+						let seed_hash = Ethash::get_seedhash(b.block().header().number());
+						to_value(&(pow_hash, seed_hash, target))
+					}
+					_ => Err(Error::invalid_params())
+				}
+			},
+			_ => Err(Error::invalid_params())
+		}
+	}
+
+	fn submit_work(&self, params: Params) -> Result<Value, Error> {
+		from_params::<(H64, H256, H256)>(params).and_then(|(nonce, pow_hash, mix_hash)| {
+//			trace!("Decoded: nonce={}, pow_hash={}, mix_hash={}", nonce, pow_hash, mix_hash);
+			let c = take_weak!(self.client);
+			let seal = vec![encode(&mix_hash).to_vec(), encode(&nonce).to_vec()];
+			let r = c.submit_seal(pow_hash, seal);
+			to_value(&r.is_ok())
+		})
+	}
+
+	fn submit_hashrate(&self, params: Params) -> Result<Value, Error> {
```

Comment:
none of this module has any tests. it would need substantial infrastructure to be able to support them.